mmcv/runner/hooks/checkpoint.py

```diff
@@ -99,3 +102,10 @@ def after_train_iter(self, runner):
         if self.sync_buffer:
             allreduce_params(runner.model.buffers())
         self._save_checkpoint(runner)
+
+    def after_run(self, runner):
+        if self.save_last:
+            if not self.out_dir:
+                self.out_dir = runner.work_dir
```

Comment:
Maybe this two lines can be moved to the init method?

Reply:
But the checkpoint hook can't access runner during `__init__`. Moved to `before_run` instead.